tools/shared.py

```diff
@@ -1077,16 +1077,21 @@ def check(value):
   return list(filter(check, values))
 
 
+SIZE_SUFFIXES = {suffix: 1024 ** i for i, suffix in enumerate(['b', 'kb', 'mb', 'gb', 'tb'])}
+
+
 def expand_byte_size_suffixes(value):
-  """Given a string with arithmetic and/or KB/MB size suffixes, such as
-  "1024*1024" or "32MB", computes how many bytes that is and returns it as an
-  integer.
+  """Given a string with KB/MB size suffixes, such as "32MB", computes how
+  many bytes that is and returns it as an integer.
   """
-  value = value.lower().replace('tb', '*1024*1024*1024*1024').replace('gb', '*1024*1024*1024').replace('mb', '*1024*1024').replace('kb', '*1024').replace('b', '')
-  try:
-    return eval(value)
-  except Exception:
-    raise Exception("Invalid byte size, valid suffixes: KB, MB, GB, TB")
+  match = re.match(r'\s*(\d+)\s*([kmgt]?b)$', value, re.I)
```

Comment:
Does this still match both lowercase and uppercase prefixes?

Reply:
Yes, that's what `re.I` is for.